src/base_alloc/base_alloc_global.c

```diff
@@ -48,10 +51,12 @@ void umf_ba_destroy_global(void) {
     }
 
     // portable version of "ba_is_initialized = UTIL_ONCE_FLAG_INIT;"
-    static UTIL_ONCE_FLAG is_initialized = UTIL_ONCE_FLAG_INIT;
-    memcpy(&ba_is_initialized, &is_initialized, sizeof(ba_is_initialized));
+    static UTIL_ONCE_FLAG set_once = UTIL_ONCE_FLAG_INIT;
+    memcpy(&ba_is_initialized, &set_once, sizeof(ba_is_initialized));
```

Comment:
`ba_is_destroyed = false;` should be added here

Reply:
Maybe the `umf-init_teardown` test fails because of this?

Reply:
could we done it in the next PR?